dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentRelationshipsHelper.java

```diff
@@ -0,0 +1,186 @@
+                        user, RESPECT_FE_ROLES));
+            } else if (inodePassed = UtilMethods.isSet(inode)) {
+                contentlets.add(this.contentletAPI.find(inode, user, RESPECT_FE_ROLES));
+            } else if (queryPassed = UtilMethods.isSet(query)) {
+                final String tmDate = (String) request.getSession().getAttribute("tm_date");
+                contentlets = ContentUtils.pull(query, offset, limit, orderBy, user, tmDate);
+            }
+        } catch (Exception e) {
+            if (idPassed) {
+                Logger.warn(this, "Can't find Content with Identifier: " + id);
+            } else if (queryPassed) {
+                Logger.warn(this, "Can't find Content with Inode: " + inode);
+            } else if (inodePassed) {
+                Logger.warn(this, "Error searching Content : " + e.getMessage());
+            }
+        }
+        return contentlets;
+    }
+
+    /**
+     * Converts the specified list of {@link Contentlet} objects to JSON format.
+     *
+     * @param contentlets The list of {@link Contentlet} objects.
+     * @return The contents as JSON.
+     * @throws JSONException    An error occurred when generating the JSON data.
+     * @throws DotDataException An error occurred when accessing the data source.
+     */
+    public String contentsAsJson(final List<Contentlet> contentlets) throws JSONException,
+            DotDataException {
+        final JSONArray resultArray = new JSONArray();
+        for (Contentlet contentlet : contentlets) {
+            final JSONObject jsonObject = toJson(contentlet);
+            final ContentletRelationships relationships = this.contentletAPI.getAllRelationships
+                    (contentlet);
+            for (ContentletRelationships.ContentletRelationshipRecords rel : relationships
+                    .getRelationshipsRecords()) {
+                JSONArray jsonArray = (JSONArray) jsonObject.optJSONArray(rel.getRelationship()
+                        .getTitle());
+                if (jsonArray == null) {
+                    jsonArray = new JSONArray();
+                }
+                for (Contentlet relatedContent : rel.getRecords()) {
+                    jsonArray.add(toJson(relatedContent));
+                }
+                jsonObject.put(rel.getRelationship().getRelationTypeValue(), jsonArray);
+            }
+            resultArray.add(jsonObject);
+        }
+        return resultArray.toString();
+    }
+
+    /**
+     * Converts the specified {@link Contentlet} to JSON format.
+     *
+     * @param contentlet The {@link Contentlet} object.
+     * @return The content as JSON.
+     * @throws JSONException    An error occurred when generating the JSON data.
+     * @throws DotDataException An error occurred when accessing the data source.
+     */
+    private JSONObject toJson(final Contentlet contentlet) throws DotDataException, JSONException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.exception.DotDataException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)